File: 3708-zigzag-grid-traversal-with-skip/3708-zigzag-grid-traversal-with-skip.py
class Solution(object):
    def zigzagTraversal(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: List[int]
        """
        n, m = len(grid), len(grid[0])
        res = []
        for i in range(n * m):
            if i % 2 == 1:
                continue
            x = i // m
            y = i % m
            if x % 2 == 1:
                y = m - y - 1
            res.append(grid[x][y])
        
        return res

File: 3708-zigzag-grid-traversal-with-skip/3708-zigzag-grid-traversal-with-skip.go
func zigzagTraversal(grid [][]int) []int {
    if len(grid) == 0 {
        return []int{}
    }
    ret := make([]int, 0)
    l := len(grid[0])
    l = l % 2
    for i := range grid {
        if i % 2 == 0 {
            for j := 0; j < len(grid[0]); j += 2 {
                ret = append(ret, grid[i][j])
            }
        } else {
            for j := len(grid[0]) - l - 1; j >= 0; j -= 2 {
                ret = append(ret, grid[i][j])
            }
        }
    }
    return ret
}